sql/api/src/main/scala/org/apache/spark/sql/types/Metadata.scala

```diff
@@ -339,6 +339,7 @@ class MetadataBuilder {
 
   private def put(key: String, value: Any): this.type = {
     map.put(key, value)
+    runtimeMap.remove(key)
```

Comment:
a small bug fix: The `map` and `runtimeMap` should be in sync always. If the key has a regular value, then it should not have an expression value in `runtimeMap` that was generated by a different value before.